.pipelines/templates/template-setup-golang-env.yml

```diff
@@ -0,0 +1,25 @@
+parameters:
+  gobin: ""
+  gopath: ""
+  goroot: ""
+  modulePath: ""
+steps:
+  - script: |
+      mkdir -p '${{ parameters.gobin }}'
+      mkdir -p '${{ parameters.gopath }}/pkg'
+      mkdir -p '${{ parameters.modulePath }}'
+      shopt -s extglob
+      shopt -s dotglob
+      mv !(gopath) '${{ parameters.modulePath }}'
+      export PATH=${{ parameters.gobin }}:$PATH
+      export PATH=${{ parameters.goroot }}/bin:$PATH
+      echo "##vso[task.prependpath]${{ parameters.gobin }}"
+      echo "##vso[task.prependpath]${{ parameters.goroot }}/bin"
+      go get -v github.com/jstemmer/go-junit-report
+      go get -v github.com/axw/gocov/gocov
+      go get -v github.com/AlekSi/gocov-xml
```

Comment:
We've previously been burned by upstream repos like these making breaking changes and causing CI outages.  We decided to try adding tools like this to /vendor to "fix" them - see the `required = [` at the top of Gopkg.toml.  If it's an easy change it might be worth doing the same here?

Reply:
sounds good